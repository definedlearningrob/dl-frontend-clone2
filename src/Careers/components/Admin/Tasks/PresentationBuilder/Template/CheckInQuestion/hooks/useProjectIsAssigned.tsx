import { useMemo } from 'react';
import { useParams } from 'react-router-dom';
import { useQuery } from '@apollo/client';
import { isEmpty } from 'lodash-es';

import teamProjectQuery from '@pbl/graphql/student/queries/teamProject';
import projectQuery from '@pbl/graphql/student/queries/project';

import { useRole } from '@shared/hooks/useRole';

export const useProjectIsAssigned = () => {
  const { isStudent } = useRole();
  const { projectId, teamId } = useParams<{ projectId: string; teamId: string }>();

  const properQuery = useMemo(() => (teamId ? teamProjectQuery : projectQuery), [teamId]);

  const queryVariables = useMemo(
    () =>
      teamId
        ? { id: projectId, teamId, track: true, trackPresentation: true }
        : { id: projectId, track: true, trackPresentation: true },
    [teamId]
  );

  const { data, error, loading } = useQuery(properQuery, {
    variables: queryVariables,
    skip: !isStudent,
  });

  if (!isStudent || loading || error) {
    return false;
  }

  return !isEmpty(data?.project?.assignedAt);
};
