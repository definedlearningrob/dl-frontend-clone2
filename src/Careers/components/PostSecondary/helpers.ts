import { InstitutionApplication } from '@dc/graphql/student/queries/institutionApplications';
import { APPLICATIONS_TYPE, RECOMMENDER_TYPE } from '@dc/resources/enums';

export const checkIfHasEnoughTeachers = (application: InstitutionApplication) => {
  const teachers = application.recommenders.filter(
    (recommender) => recommender.type === RECOMMENDER_TYPE.TEACHER
  );

  return teachers.length >= (application.institution.minTeacherEval || 0);
};

export const checkIfCanBeMovedToCommonApp = (
  application: InstitutionApplication,
  hasAccountConnected: boolean
) =>
  application.type === APPLICATIONS_TYPE.DIRECT &&
  application.institution.commonAppEnabled &&
  hasAccountConnected;
