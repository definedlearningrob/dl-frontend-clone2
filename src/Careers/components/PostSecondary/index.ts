export * from './Search/Search';
export * from './FutureChecklist/FutureChecklist';
export * from './Applications/Applications';
export * from './Favorites/Favorites';
export * from './EmptyFutureChecklist/EmptyFutureChecklist';
export * from './EmptyFavorites/EmptyFavorites';
export * from './FavoritesList/FavoritesList';
export * from './FavoritesCarousel/FavoritesCarousel';
export * from '@dc/components/PostSecondary/ApplicationRecommenderList/ApplicationRecommenderList';
export * from './EmptyApplications/EmptyApplications';
export * from '@dc/components/PostSecondary/ApplicationDetailsModal/ApplicationDetailsModal';
export * from './PostSecondaryContent';
