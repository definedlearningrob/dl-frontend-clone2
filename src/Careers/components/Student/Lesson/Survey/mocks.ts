import lessonInCourseQuery from '@dc/graphql/student/queries/lessonInCourse';
import { CREATE_CAREER_REVIEW_SURVEY_ATTEMPT } from '@dc/graphql/student/mutations/createCareerReviewSurveyAttempt';
import createCareerReviewSurveyAnswersMutation from '@dc/graphql/student/mutations/createCareerReviewSurveyAnswers';
import currentCoursesQuery from '@dc/graphql/student/queries/currentCourses';
import { SURVEY_QUESTION_TYPE } from '@dc/components/Student/Lesson/Survey/types';

export const createAttemptSpy = jest.fn();
export const lessonInCourseSpy = jest.fn();

export let createCareerReviewSurveyAnswersSpy = jest.fn();
export let currentCoursesSpy = jest.fn();

export const mocks = [
  {
    request: {
      query: lessonInCourseQuery,
      variables: { courseId: '2', lessonId: '111', track: true },
    },
    result: () => {
      lessonInCourseSpy();

      return {
        data: {
          course: {
            careerName: 'Dental Laboratory Technicians',
            content: {
              checkIns: [],
              items: [],
            },
            hasInstitutionsInStudentState: true,
            id: '2',
            lesson: {
              assignments: [],
              attachments: [],
              careerReviewSurvey: {
                version: 1,
                performed: false,
                questions: [
                  {
                    answer: [],
                    id: '1',
                    options: [
                      {
                        option: 'Yes, definitely',
                        step: 1,
                      },
                      {
                        option: 'Definitely not',
                        step: 2,
                      },
                    ],
                    question: 'Can blind people see their dreams?',
                    type: 'single_choice' as SURVEY_QUESTION_TYPE,
                  },
                  {
                    answer: [],
                    id: '2',
                    options: [
                      {
                        option: 'Very Interested',
                        step: 1,
                      },
                      {
                        option: 'Not at all Interested',
                        step: 2,
                      },
                      {
                        option: 'Other',
                        step: 3,
                      },
                    ],
                    question: 'How interested are you in the Pathway this course is a part of?',
                    type: 'multiple_choice' as SURVEY_QUESTION_TYPE,
                  },
                ],
                __typename: 'CareerReviewSurvey',
              },
              checkInGroups: [],
              checkInQuestions: [],
              description: {
                introduction: null,
                goal: null,
                role: null,
                audience: null,
                situation: null,
                __typename: 'LessonDescription',
              },
              hasPresentation: false,
              id: '609',
              imageUrl: 'http://test.com/image.png',
              name: 'Career Review Survey',
              externalPresentations: [],
              researchLinks: [],
              texts: [],
              type: 'career_review_survey',
              videos: [],
              vocabularies: [],
              __typename: 'Lesson',
            },
            lessons: [
              {
                id: '782',
                step: 4,
                careerReviewSurvey: null,
              },
              {
                id: '609',
                step: 5,
                careerReviewSurvey: {
                  performed: false,
                  version: 1,
                  __typename: 'CareerReviewSurvey',
                },
              },
              {
                id: '275',
                step: 1,
                careerReviewSurvey: null,
              },
              {
                id: '331',
                step: 2,
                careerReviewSurvey: null,
              },
              {
                id: '668',
                step: 3,
                careerReviewSurvey: null,
              },
            ],
            name: 'Dental Laboratory Technician',
            progress: {
              submitted: 10,
              total: 10,
            },
            reviewSurvey: {
              questions: [
                {
                  id: '1',
                },
                {
                  id: '2',
                },
                {
                  id: '3',
                },
                {
                  id: '4',
                },
                {
                  id: '5',
                },
                {
                  id: '6',
                },
                {
                  id: '7',
                },
                {
                  id: '8',
                },
                {
                  id: '9',
                },
                {
                  id: '10',
                },
                {
                  id: '11',
                },
              ],
            },
          },
        },
      };
    },
  },
  {
    request: {
      query: CREATE_CAREER_REVIEW_SURVEY_ATTEMPT,
      variables: { input: { contextType: 'COURSE', contextId: '1' } },
    },
    result: () => {
      createAttemptSpy();

      return {
        data: {
          createCareerReviewSurveyAttempt: {
            careerReviewSurveyAttempt: {
              id: '2',
              status: 'IN_PROGRESS',
            },
          },
        },
      };
    },
  },
  {
    request: {
      query: createCareerReviewSurveyAnswersMutation,
      variables: {
        input: {
          contextId: '1',
          contextType: 'COURSE',
          answers: [
            { questionId: '1', answer: ['Yes, definitely'] },
            { questionId: '2', answer: ['Very Interested'] },
          ],
        },
      },
    },
    result: () => {
      createCareerReviewSurveyAnswersSpy();

      return {
        data: {
          createCareerReviewSurveyAnswers: {
            status: 'success',
          },
        },
      };
    },
  },
  {
    request: {
      query: currentCoursesQuery,
    },
    result: () => {
      currentCoursesSpy();

      return {
        data: {
          currentCourses: {
            id: '1',
            imageUrl: '',
            name: '',
            progress: {
              submitted: 2,
              total: 5,
            },
            status: 'success',
            pathway: {
              name: '',
            },
            thumbnailUrl: '',
            type: 'survey',
          },
        },
      };
    },
  },
];
