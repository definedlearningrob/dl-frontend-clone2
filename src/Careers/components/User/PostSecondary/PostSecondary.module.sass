@import '@shared/stylesheets/config.sass'

$common-icon-width: 118px
$info-max-width: 688px

.container
  margin: 0 auto

  +media-breakpoint-up(xxxl)
    max-width: 75%

.info
  max-width: $info-max-width
  text-align: center
  font-weight: $font-weight-regular
  line-height: $line-height-lg
  font-size: $font-size-base
  margin: 0 auto
