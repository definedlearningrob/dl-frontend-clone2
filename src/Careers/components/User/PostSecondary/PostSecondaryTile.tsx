import cx from 'classnames';
import { FC, SVGProps } from 'react';
import { useTranslation } from 'react-i18next';

import { IconContainer } from '@shared/components/IconContainer/IconContainer';
import { Badge } from '@shared/components/Badge/Badge';
import { ReactComponent as LockedIcon } from '@shared/svg/padlock.svg';

type Props = {
  Icon: FC<SVGProps<SVGSVGElement>>;
  title: string;
  withCustomImage?: boolean;
  disabled?: boolean;
};

export const PostSecondaryTile = ({ Icon, title, withCustomImage, disabled }: Props) => {
  const { t } = useTranslation();

  const customImageClasses = cx('flex-grow h-lg', { '[&_path]:fill-neutral-600': disabled });
  const iconWrapperClasses = cx('bg-neutral-200 rounded-sm', {
    'bg-transparent w-full': withCustomImage,
  });
  const tileClasses = cx(
    'flex flex-col justify-center items-center py-lg px-md gap-md bg-white rounded-sm',
    {
      'relative border border-neutral-300 text-neutral-600': disabled,
    }
  );

  return (
    <div className={tileClasses}>
      {disabled && (
        <Badge className='absolute top-sm left-sm flex gap-xxs items-center' type='neutral'>
          <IconContainer Icon={LockedIcon} paddingSize='none' size='sm' />
          {t('user.postSecondary.commonAppRequests.locked')}
        </Badge>
      )}
      <div className={iconWrapperClasses}>
        {withCustomImage ? (
          <div className='flex'>
            <Icon className={customImageClasses} />
          </div>
        ) : (
          <IconContainer Icon={Icon} className='text-neutral-800' paddingSize='xs' size='md' />
        )}
      </div>
      <h4 className='m-0 text-base leading-base font-bold text-center'>{title}</h4>
    </div>
  );
};
