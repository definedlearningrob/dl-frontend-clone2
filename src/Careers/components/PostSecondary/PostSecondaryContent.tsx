import cx from 'classnames';
import { useTranslation } from 'react-i18next';

import { Applications, Favorites, FutureChecklist, Search } from '@dc/components/PostSecondary/';

import Link from '@shared/components/Link';
import { useBreakpointUp } from '@shared/hooks/useBreakpointUp';

import styles from './PostSecondaryContent.module.sass';

type Props = { applicationsEnabled: boolean };

export const PostSecondaryContent = ({ applicationsEnabled }: Props) => {
  const { t } = useTranslation();
  const contentClassName = cx(styles.content, {
    [styles.applicationsEnabled]: applicationsEnabled,
  });
  const isFullHD = useBreakpointUp({ breakpoint: 'xxxl' });

  const favoritesSectionClassname = cx(styles.favoritesSection, {
    [styles.verticalList]: !applicationsEnabled,
  });

  return (
    <div className={contentClassName}>
      <div className={styles.searchSection}>
        <Search />
      </div>
      <div className={styles.futureListSection}>
        <FutureChecklist showAdditionalContent={applicationsEnabled} />
      </div>
      <div className={favoritesSectionClassname}>
        <Favorites isCarousel={applicationsEnabled} />
      </div>
      {applicationsEnabled && (
        <div className={styles.applicationsSection}>
          <Applications
            actions={
              <Link
                size={isFullHD ? 'md' : 'sm'}
                to='/post-secondary/manage-applications'
                variant='primary-outlined'>
                {t('student.postSecondary.applicationsSection.manageApplications')}
              </Link>
            }
          />
        </div>
      )}
    </div>
  );
};
