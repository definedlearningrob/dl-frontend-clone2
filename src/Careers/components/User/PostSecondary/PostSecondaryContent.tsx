import { useTranslation } from 'react-i18next';
import { useToggle } from 'react-use';
import { Link } from 'react-router-dom';
import cx from 'classnames';

import useUserInfo from '@dc/hooks/useUserInfo';
import { TUserInfo } from '@dc/graphql/user/queries/userInfo';
import { PostSecondaryTile } from '@dc/components/User/PostSecondary/PostSecondaryTile';
import { PostSecondaryImpersonateModal } from '@dc/components/User/PostSecondaryImpersonateModal/PostSecondaryImpersonateModal';

import { ReactComponent as AppIcon } from '@shared/assets/icons/common_app.svg';
import { ReactComponent as SearchIcon } from '@shared/assets/icons/search.svg';
import { ReactComponent as ImpersonateIcon } from '@shared/assets/icons/visible_primary.svg';

import styles from './PostSecondary.module.sass';

export const PostSecondaryContent = () => {
  const { t } = useTranslation();
  const { userInfo } = useUserInfo<TUserInfo>();
  const [isImpersonateModalOpen, toggleImpersonateModalOpen] = useToggle(false);

  const { hasRecommenderInvitation } = userInfo.commonAppData;
  const isCounselor = userInfo.permissions.counselor;
  const { canImpersonate } = userInfo.permissions;
  const hasAccessToImpersonate = !userInfo.isImpersonated && canImpersonate && isCounselor;

  const hasAccessToCommonApp = userInfo.entities.nodes.some(
    (entity) => entity.settings.postSecondaryApplicationsEnabled
  );
  const dashboardInfoKey = hasRecommenderInvitation
    ? 'user.postSecondary.invitedRecommenderDashboardInfo'
    : 'user.postSecondary.counselorDashboardInfo';

  const shouldDisplayCommonAppTile =
    hasAccessToCommonApp && (hasRecommenderInvitation || isCounselor);

  return (
    <div className={styles.container}>
      <div className='text-neutral-800 py-md px-0'>
        <h3 className='font-bold leading-base text-lg text-center mb-sm'>
          {t('user.postSecondary.dashboardTitle')}
        </h3>
        <p className={styles.info}>{t(dashboardInfoKey)}</p>
      </div>
      <div className='flex justify-center items-start gap-base xxxl:gap-md'>
        {shouldDisplayCommonAppTile && (
          <div>
            <Link
              aria-disabled={!hasRecommenderInvitation}
              className={cx({ 'pointer-events-none': !hasRecommenderInvitation })}
              to='/post-secondary/commonapp-requests'>
              <PostSecondaryTile
                Icon={AppIcon}
                disabled={!hasRecommenderInvitation}
                title={t('user.postSecondary.commonAppRequests.title')}
                withCustomImage={true}
              />
            </Link>
            {!hasRecommenderInvitation && (
              <div className='flex mt-sm'>
                <p className='flex-grow w-0 text-xxs leading-lg xxxl:text-xs'>
                  {t('user.postSecondary.commonAppRequestsDisabled')}
                </p>
              </div>
            )}
          </div>
        )}
        <Link to='/post-secondary/search'>
          <PostSecondaryTile Icon={SearchIcon} title={t('user.postSecondary.search')} />
        </Link>
        {hasAccessToImpersonate && (
          <button className='text-primary-500' onClick={toggleImpersonateModalOpen}>
            <PostSecondaryTile
              Icon={ImpersonateIcon}
              title={t('user.postSecondary.loginAsCounselor')}
            />
          </button>
        )}
      </div>
      {isImpersonateModalOpen && (
        <PostSecondaryImpersonateModal onCloseModal={toggleImpersonateModalOpen} />
      )}
    </div>
  );
};
