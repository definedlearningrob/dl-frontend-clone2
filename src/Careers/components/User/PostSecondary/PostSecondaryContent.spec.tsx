import { MockedProvider } from '@apollo/client/testing';
import { Route } from 'react-router-dom';
import { screen, waitFor } from '@testing-library/react';
import userEvent from '@testing-library/user-event';

import { PostSecondaryContent } from '@dc/components/User/PostSecondary/PostSecondaryContent';
import { renderWithRouterAndReduxProvider } from '@dc/utils/test';
import { UserInfoProvider } from '@dc/hooks/useUserInfo';
import { userInfoMock } from '@dc/tests/mocks/userMocks';
import { TUserInfo } from '@dc/graphql/user/queries/userInfo';
import { SYNC_STATUS } from '@dc/resources/enums';
import { COUNSELORS } from '@dc/graphql/user/queries/counselors';

const getAllCounselorsFetchSpy = jest.fn();

const getAllCounselorsMock = {
  request: {
    query: COUNSELORS,
    variables: { perPage: 100 },
  },
  result() {
    getAllCounselorsFetchSpy();

    return {
      data: {
        counselors: {
          pagesCount: 1,
          nodesCount: 2,
          nodes: [
            {
              firstName: 'Cindy',
              fullName: 'Cindy Admino',
              lastName: 'Admino',
              uuid: 'b6f5217c-451a-4745-96ff-7aa54ba970a8',
              __typename: 'User',
            },
            {
              firstName: 'Bruce',
              fullName: 'Bruce Wayne',
              lastName: 'Wayne',
              uuid: 'b6f5217c-451a-4745-96ff-7aa54ba970a8',
              __typename: 'User',
            },
            {
              firstName: 'John',
              fullName: 'John Doe',
              lastName: 'Doe',
              uuid: 'b6f5217c-451a-4745-96ff-7aa54ba970a8',
            },
            {
              firstName: 'Jane',
              fullName: 'Jane Doe',
              lastName: 'Doe',
              uuid: 'b6f5217c-451a-4745-96ff-7aa54ba970a8',
            },
          ],
        },
      },
    };
  },
};

const renderPostSecondaryContent = (userInfoOverride: Partial<TUserInfo> = {}) =>
  renderWithRouterAndReduxProvider(
    <MockedProvider mocks={[getAllCounselorsMock]}>
      <UserInfoProvider
        value={{
          userInfo: {
            ...userInfoMock.result.data.userInfo,
            ...userInfoOverride,
          },
        }}>
        <Route path='/post-secondary'>
          <PostSecondaryContent />
        </Route>
      </UserInfoProvider>
    </MockedProvider>,
    { route: '/post-secondary' }
  );

describe('PostSecondaryContent', () => {
  it('should render correctly', async () => {
    const { container } = renderPostSecondaryContent();
    const searchTile = await screen.findByText('Postsecondary Search');

    expect(searchTile).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });

  it('should render Common App Requests Post Secondary Impersonate counselor tile', async () => {
    renderPostSecondaryContent({
      firstName: 'Jack',
      lastName: 'Sparrow',
      commonAppData: {
        hasRecommenderInvitation: true,
        hasTeacherInvitation: false,
        hasCounselorInvitation: false,
        hasCounselorProfileFormCompleted: false,
        hasTeacherProfileFormCompleted: false,
        syncStatus: {
          lastSyncedAt: '2022-03-11T15:00:00.000Z',
          status: SYNC_STATUS.COMPLETED,
        },
      },
      entities: {
        nodes: [
          {
            uuid: 'entityuuid',
            settings: {
              classManagementEnabled: false,
              postSecondaryApplicationsEnabled: true,
              schoolYearStartDate: {
                day: 1,
                month: 9,
              },
            },
            reportTypes: [],
          },
        ],
      },
      permissions: {
        counselor: true,
        wblAdmin: false,
        canImpersonate: true,
        canBrowseReports: false,
      },
    });

    const commonAppRequestsTile = await screen.findByText('Common App Requests');
    const impersonateCounselorTile = await screen.findByText('Login as Counselor');
    const postSecondarySearchTile = await screen.findByText('Postsecondary Search');

    expect(postSecondarySearchTile).toBeInTheDocument();
    expect(impersonateCounselorTile).toBeInTheDocument();
    expect(commonAppRequestsTile).toBeInTheDocument();
  });

  it('should allow to impersonate a counselor', async () => {
    renderPostSecondaryContent({
      firstName: 'Jack',
      lastName: 'Sparrow',
      commonAppData: {
        hasRecommenderInvitation: true,
        hasTeacherInvitation: false,
        hasCounselorInvitation: false,
        hasCounselorProfileFormCompleted: false,
        hasTeacherProfileFormCompleted: false,
        syncStatus: {
          lastSyncedAt: '2022-03-11T15:00:00.000Z',
          status: SYNC_STATUS.COMPLETED,
        },
      },
      permissions: {
        counselor: true,
        wblAdmin: false,
        canImpersonate: true,
        canBrowseReports: false,
      },
    });

    const impersonateCounselorTile = await screen.findByText('Login as Counselor');
    userEvent.click(impersonateCounselorTile);

    await waitFor(() => {
      expect(getAllCounselorsFetchSpy).toHaveBeenCalled();
    });

    const impersonateModal = await screen.findByRole('dialog');

    expect(impersonateModal).toHaveTextContent('Select counselor you want to login as');
    expect(impersonateModal).toHaveTextContent(
      'Select the counselor you want to login as so you can manage their content in Postsecondary'
    );
    const dropDownList = await screen.findByRole('combobox');
    userEvent.click(dropDownList);

    expect(await screen.findByText('Cindy Admino')).toBeInTheDocument();
  });

  it('should display disabled Common App Requests tile when counselor has no invitations', async () => {
    const { container } = renderPostSecondaryContent({
      commonAppData: {
        hasRecommenderInvitation: false,
        hasTeacherInvitation: false,
        hasCounselorInvitation: false,
        hasCounselorProfileFormCompleted: false,
        hasTeacherProfileFormCompleted: false,
        syncStatus: {
          lastSyncedAt: '2022-03-11T15:00:00.000Z',
          status: SYNC_STATUS.COMPLETED,
        },
      },
      entities: {
        nodes: [
          {
            uuid: 'entityuuid',
            settings: {
              classManagementEnabled: false,
              postSecondaryApplicationsEnabled: true,
              schoolYearStartDate: {
                day: 1,
                month: 9,
              },
            },
            reportTypes: [],
          },
        ],
      },
      permissions: {
        counselor: true,
        wblAdmin: false,
        canImpersonate: false,
        canBrowseReports: false,
      },
    });

    const commonAppRequestsTile = await screen.findByRole('link', {
      name: 'Locked Common App Requests',
    });

    expect(commonAppRequestsTile).toHaveAttribute('aria-disabled', 'true');
    expect(commonAppRequestsTile).toHaveClass('pointer-events-none');
    expect(
      screen.getByText(
        'You currently have no Common App invites. The tile will be enabled when a student invites you.'
      )
    ).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });

  it('should not display common app requests tile when entity has no access to it', async () => {
    const { container } = renderPostSecondaryContent({
      commonAppData: {
        hasRecommenderInvitation: false,
        hasTeacherInvitation: false,
        hasCounselorInvitation: false,
        hasCounselorProfileFormCompleted: false,
        hasTeacherProfileFormCompleted: false,
        syncStatus: {
          lastSyncedAt: '2022-03-11T15:00:00.000Z',
          status: SYNC_STATUS.COMPLETED,
        },
      },
      permissions: {
        counselor: true,
        wblAdmin: false,
        canImpersonate: false,
        canBrowseReports: false,
      },
    });

    const postSecondarySearchTile = await screen.findByText('Postsecondary Search');

    expect(postSecondarySearchTile).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });
});
