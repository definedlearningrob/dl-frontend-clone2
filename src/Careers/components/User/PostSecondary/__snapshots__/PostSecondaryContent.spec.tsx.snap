// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PostSecondaryContent should display disabled Common App Requests tile when counselor has no invitations 1`] = `
<div>
  <div
    class="container"
  >
    <div
      class="text-neutral-800 py-md px-0"
    >
      <h3
        class="font-bold leading-base text-lg text-center mb-sm"
      >
        College & Future Planning
      </h3>
      <p
        class="info"
      >
        Select an option to view available futures or manage your students
      </p>
    </div>
    <div
      class="flex justify-center items-start gap-base xxxl:gap-md"
    >
      <div>
        <a
          aria-disabled="true"
          class="pointer-events-none"
          href="/post-secondary/commonapp-requests"
        >
          <div
            class="flex flex-col justify-center items-center py-lg px-md gap-md bg-white rounded-sm relative border border-neutral-300 text-neutral-600"
          >
            <div
              class="font-medium leading-lg flex items-center absolute top-sm left-sm flex gap-xxs items-center bg-neutral-200 text-font-secondary rounded-sm p-xxs text-xs gap-xxs"
            >
              <div
                class="flex"
              >
                <div
                  class="flex align-center justify-center grow-0 shrink-0"
                  data-testid="icon"
                >
                  <div
                    class="h-sm w-sm"
                  />
                </div>
              </div>
              Locked
            </div>
            <div
              class="bg-neutral-200 rounded-sm bg-transparent w-full"
            >
              <div
                class="flex"
              >
                <div
                  class="flex-grow h-lg [&_path]:fill-neutral-600"
                />
              </div>
            </div>
            <h4
              class="m-0 text-base leading-base font-bold text-center"
            >
              Common App Requests
            </h4>
          </div>
        </a>
        <div
          class="flex mt-sm"
        >
          <p
            class="flex-grow w-0 text-xxs leading-lg xxxl:text-xs"
          >
            You currently have no Common App invites. The tile will be enabled when a student invites you.
          </p>
        </div>
      </div>
      <a
        href="/post-secondary/search"
      >
        <div
          class="flex flex-col justify-center items-center py-lg px-md gap-md bg-white rounded-sm"
        >
          <div
            class="bg-neutral-200 rounded-sm"
          >
            <div
              class="flex"
            >
              <div
                class="flex align-center justify-center grow-0 shrink-0 text-neutral-800 p-xs"
                data-testid="icon"
              >
                <div
                  class="h-md w-md"
                />
              </div>
            </div>
          </div>
          <h4
            class="m-0 text-base leading-base font-bold text-center"
          >
            Postsecondary Search
          </h4>
        </div>
      </a>
    </div>
  </div>
</div>
`;

exports[`PostSecondaryContent should not display common app requests tile when entity has no access to it 1`] = `
<div>
  <div
    class="container"
  >
    <div
      class="text-neutral-800 py-md px-0"
    >
      <h3
        class="font-bold leading-base text-lg text-center mb-sm"
      >
        College & Future Planning
      </h3>
      <p
        class="info"
      >
        Select an option to view available futures or manage your students
      </p>
    </div>
    <div
      class="flex justify-center items-start gap-base xxxl:gap-md"
    >
      <a
        href="/post-secondary/search"
      >
        <div
          class="flex flex-col justify-center items-center py-lg px-md gap-md bg-white rounded-sm"
        >
          <div
            class="bg-neutral-200 rounded-sm"
          >
            <div
              class="flex"
            >
              <div
                class="flex align-center justify-center grow-0 shrink-0 text-neutral-800 p-xs"
                data-testid="icon"
              >
                <div
                  class="h-md w-md"
                />
              </div>
            </div>
          </div>
          <h4
            class="m-0 text-base leading-base font-bold text-center"
          >
            Postsecondary Search
          </h4>
        </div>
      </a>
    </div>
  </div>
</div>
`;

exports[`PostSecondaryContent should render correctly 1`] = `
<div>
  <div
    class="container"
  >
    <div
      class="text-neutral-800 py-md px-0"
    >
      <h3
        class="font-bold leading-base text-lg text-center mb-sm"
      >
        College & Future Planning
      </h3>
      <p
        class="info"
      >
        Select an option to view available futures or manage your students
      </p>
    </div>
    <div
      class="flex justify-center items-start gap-base xxxl:gap-md"
    >
      <a
        href="/post-secondary/search"
      >
        <div
          class="flex flex-col justify-center items-center py-lg px-md gap-md bg-white rounded-sm"
        >
          <div
            class="bg-neutral-200 rounded-sm"
          >
            <div
              class="flex"
            >
              <div
                class="flex align-center justify-center grow-0 shrink-0 text-neutral-800 p-xs"
                data-testid="icon"
              >
                <div
                  class="h-md w-md"
                />
              </div>
            </div>
          </div>
          <h4
            class="m-0 text-base leading-base font-bold text-center"
          >
            Postsecondary Search
          </h4>
        </div>
      </a>
    </div>
  </div>
</div>
`;
